backend/dataall/aws/handlers/lakeformation.py

```diff
@@ -13,26 +13,25 @@ def __init__(self):
         pass
 
     @staticmethod
-    def describe_resource(resource_arn, role_arn, accountid, region):
+    def check_existing_lf_registered_location(resource_arn, accountid, region):
         """
-        Describes a LF data location
+        Checks if there is a non-dataall-created registered location for the Dataset
+        Returns False is already existing location else return the resource info
         """
         try:
             session = SessionHelper.remote_session(accountid)
             lf_client = session.client('lakeformation', region_name=region)
-
             response = lf_client.describe_resource(ResourceArn=resource_arn)
-
-            log.info(f'LF data location already registered: {response}, checking if data.all registered it ...')
-            if response['ResourceInfo']['RoleArn'] == role_arn:
+            registered_role_name = response['ResourceInfo']['RoleArn'].lstrip(f"arn:aws:iam::{accountid}:role/")
+            pivot_role_name = SessionHelper.get_delegation_role_name()
+            log.info(f'LF data location already registered: {response}, registered with role {registered_role_name}')
+            if registered_role_name[:11] == pivot_role_name[:11]:
```

Comment:
Why 11? The default name is `dataallPivotRole` and we can also have `dataallPivotRole-cdk`. 
+ Let's create a constant with the proper name for this

Reply:
Miscounted that one, but yes you are right it better be in a constant